modules/cdc-ext/src/test/java/org/apache/ignite/cdc/AbstractReplicationTest.java

```diff
@@ -0,0 +1,331 @@
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.ignite.cache.CacheAtomicityMode.ATOMIC;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.cluster.ClusterState.ACTIVE;
+import static org.apache.ignite.spi.communication.tcp.TcpCommunicationSpi.DFLT_PORT_RANGE;
+import static org.apache.ignite.testframework.GridTestUtils.runAsync;
+import static org.apache.ignite.testframework.GridTestUtils.waitForCondition;
+
+/** */
+@RunWith(Parameterized.class)
+public abstract class AbstractReplicationTest extends GridCommonAbstractTest {
+    /** Cache mode. */
+    @Parameterized.Parameter
+    public CacheAtomicityMode cacheMode;
+
+    /** */
+    @Parameterized.Parameter(1)
+    public int backupCnt;
+
+    /** @return Test parameters. */
+    @Parameterized.Parameters(name = "cacheMode={0},backupCnt={1}")
+    public static Collection<?> parameters() {
+        List<Object[]> params = new ArrayList<>();
+
+        for (CacheAtomicityMode mode : EnumSet.of(ATOMIC, TRANSACTIONAL))
+            for (int i = 0; i < 2; i++)
+                params.add(new Object[] {mode, i});
+
+        return params;
+    }
+
+    /** */
+    public static final String AP_CACHE = "active-passive-cache";
+
+    /** */
+    public static final String ACTIVE_ACTIVE_CACHE = "active-active-cache";
+
+    /** */
+    public static final byte SRC_CLUSTER_ID = 26;
+
+    /** */
+    public static final byte DEST_CLUSTER_ID = 27;
+
+    /** */
+    public static final int EXISTS = 1;
+
+    /** */
+    public static final int REMOVED = 2;
+
+    /** */
+    public static final int KEYS_CNT = 50;
+
+    /** */
+    protected static IgniteEx[] srcCluster;
+
+    /** */
+    protected static IgniteConfiguration[] srcClusterCliCfg;
+
+    /** */
+    protected static IgniteEx[] destCluster;
+
+    /** */
+    protected static IgniteConfiguration[] destClusterCliCfg;
+
+    /** */
+    private int commPort = TcpCommunicationSpi.DFLT_PORT;
+
+    /** */
+    private int discoPort = TcpDiscoverySpi.DFLT_PORT;
+
+    /** */
+    private byte clusterId = SRC_CLUSTER_ID;
+
+    /** */
+    protected int clientsCnt = 1;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName)
+            .setDiscoverySpi(new TcpDiscoverySpi()
+                .setLocalPort(discoPort)
+                .setIpFinder(new TcpDiscoveryVmIpFinder() {{
+                    setAddresses(Collections.singleton("127.0.0.1:" + discoPort + ".." + (discoPort + DFLT_PORT_RANGE)));
+                }}))
+            .setCommunicationSpi(new TcpCommunicationSpi()
+                .setLocalPort(commPort));
```

Comment:
Commented this, but tests still green.

Reply:
https://ignite.apache.org/docs/latest/clustering/tcp-ip-discovery#isolated-clusters-on-same-set-of-machines - I've followed the documentation to setup isolated clusters.